### DISCUSSION TITLE
Pod PortForward using HTTPS

### DISCUSSION THREAD

**Discussion by tiagolobocastro** [kube-rs/kube#947] [Q&A]
Hi, I'm looking at the pod portforward again and I've got another question if I may. Does the current port forward implementation support HTTPS?
I've tried using [this](https://github.com/kube-rs/kube-rs/blob/master/examples/pod_portforward_bind.rs) example and sending requests over curl, but it doesn't seem to work. On the other hand, if I create the bind using the kubectl binary it seems to work.

Off-topic: since performance is not too much of an issue, I'm also thinking about using the apiserver proxy instead. This would potentially solve the issue of having an "outdated port forward" if the pod we're connected to is deleted? Anyone got any experience or gotchas with this approach?

Thanks!

**Comment by nightkr** [comment]
Odd, looks like the example listed runs its own HTTP server and forwards the requests one by one (like a reverse HTTP proxy). Ideally it should just be a "dumb" TCP proxy (which is what `kubectl` does).

**Reply by nightkr** [reply]
For what it's worth, `kubectl proxy` wouldn't help here, since that's about proxying requests to the API server (rather than to services running inside of the cluster).

**Reply by tiagolobocastro** [reply]
I was thinking of using this: "/api/v1/namespaces/xxxxx/services/https:service:https/proxy/xxxxx". One downside I've noticed is that the "Authorization" header gets stripped, so I'd need to supporting passing JWT through a different header name.
